tests/holder/test_events.py

```diff
@@ -102,6 +115,40 @@ async def test_pres_req_received_sent_on_state(profile, mock_send_to_admins):
     assert isinstance(mock_send_to_admins.message, message)
 
 
+send_condition_parameters = [
+    ("all-credentials", {}, {"example:123", "example:456", "example:789"}, True),
+    (
+        "",
+        {"example:123", "example:456", "example:789"},
+        {"example:123", "example:456", "example:789"},
+        False,
+    ),
+    (
+        "limited-credentials",
+        {"example:123", "example:456"},
+        {"example:123", "example:456", "example:789"},
+        True,
+    ),
+    (
+        "limited-credentials",
+        {"example:123", "example:456", "example:789"},
+        {"example:123", "example:456"},
+        False,
+    ),
+]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "privilege, record_cred_def_id, approved, result", send_condition_parameters
+)
+async def test_send_condition(privilege, record_cred_def_id, approved, result):
```

Comment:
We're missing principal in this parameter list, I think

Reply:
Also missing record in parameter list?

Reply:
Is it not sufficient to pass in `privilege` and `record_cred_def_id` as parameters, which I then set equal to `principal.privileges` and `record.credential_definition_id`, respectively?

Reply:
Without passing them in, the `principal` used in the test method would actually be the method `principal` defined above. Since python doesn't mind setting arbitrary values on just about anything, it's not complaining but is of course a bit off semantically. Same for the record.